Description: mount-image-callback: update partitions and settle on nbd mounts.
 After mount-image-callback sets up a nbd device, it needs to call
 'blockdev --rereadpt' and 'udevadm settle' to ensure that the udev
 events are created and handled for this new block device.
Ubuntu-bug: https://launchpad.net/bugs/1741300
Applied-upstream: commit: revno 319
--- a/bin/mount-image-callback
+++ b/bin/mount-image-callback
@@ -304,6 +304,14 @@ mount_callback_umount() {
 		local nptnum=""
 		debug 1 "connected $img_in ($fmt) to $nbd. waiting for device."
 		i=0
+
+		local out=""
+		# This can fail due to udev events, but we ignore that.  We need to ensure
+		# it happens for where it doesnt happen automatically (LP: #1741300)
+		out=$(blockdev --rereadpt "$nbd" 2>&1) ||
+			debug 1 "blockdev rereadpt $nbd failed"
+		udevadm settle
+
 		while i=$(($i+1)):; do
 			get_partition "$nbd" && nptnum="$_RET" && break
 			[ $i -eq 40 ] && {
